### ISSUE TITLE
[Feature] 🎨 Color Picker 🎨

### ISSUE THREAD

**Issue by ryanraposo** [ryanraposo/codeui#24]
Using an external color picker is annoying and VS Code's built-in picker is restricted to text editing:

![the-text-editor-way](https://user-images.githubusercontent.com/38861352/104355359-8ec6e280-54d8-11eb-8b5e-cabf5ae4b5b5.png)

### Priorities:

- Implement a color picker view
- Make every attempt to mirror VS Code's overall design and style.
- Respond to user activity (ex: color-picker updates to reflect the color of a selected element).
- Add CodeUI extension settings/commands where appropriate

### Additional context

Use `color-picker` branch for work on this feature.

### Updates

- (12/1/21): The color picker now updates to reflect selection in the Elements view. Thinking about a reversed, toggle-able, "paint" mode where you can apply the picker's colour to elements just by clicking on them.

![picker-updates-crop](https://user-images.githubusercontent.com/38861352/104337752-d1ca8b00-54c3-11eb-8e92-14e5603160cb.gif)

- (12/10/2020) :  New approach. Added a statusBarItem that updates according to the hex and hue of the selection. Triggers copy-to-clipboard command on click. Removes any need to clutter the view with more widgets. Also, looks pretty cool. 

![status-bar-color](https://user-images.githubusercontent.com/38861352/101744311-ed8fdb00-3a98-11eb-91b5-0f2d274084ca.gif)

- (12/09/2020): Simple copy-to-clipboard functionality added. Not the minimum-viable-product we need, but a start. Very interested in help/feedback. Two rules, though! 1) no frumpy webview abuse and 2) keep it simple :)

- (11/27/2020) : A new view 'Color' with a picker has been added. No way to access the selection yet.

![colors-screenshot](https://user-images.githubusercontent.com/38861352/100413746-17331600-3046-11eb-9c3f-86cec0b785fa.png)

**Comment by ryanraposo** [comment]
paging local expert on the exciting & ethical application of cutting-edge Webview technology, @AssisrMatheus

**Comment by AssisrMatheus** [comment]
Neeeat! That is indeed a great feature, feel free to ping me for help/questions regarding anything, webview or not!

**Comment by ryanraposo** [comment]
> Neeeat! That is indeed a great feature, feel free to ping me for help/questions regarding anything, webview or not!

Perfect, thank you 👍

**Comment by ryanraposo** [comment]
Progress on `color-picker` branch:

![colors-screenshot](https://user-images.githubusercontent.com/38861352/100293796-d4dcdc80-2f52-11eb-8e06-41f979fb8e3c.png)

**Comment by ryanraposo** [comment]
@AdamPhilipSmith hey! Check out #25, #26, and #27. If you need help or have any questions, just tag me in a comment and we'll work it out!

**Comment by ryanraposo** [comment]
(12/10/2020) :  New approach. Added a statusBarItem that updates according to the hex and hue of the selection. Triggers copy-to-clipboard command on click. Removes any need to clutter the view with more widgets. Also, looks pretty cool. 

![status-bar-color](https://user-images.githubusercontent.com/38861352/101744311-ed8fdb00-3a98-11eb-91b5-0f2d274084ca.gif)

**Comment by ryanraposo** [comment]
@AssisrMatheus down to help with something? I'm wondering about resizing/adjusting that might be necessary when the Color view container is an awkward size, clipping or dwarfing the Color Wheel. Do you know what I mean? How would you handle it? Would you handle it? Curious about your thoughts from the UX decision down to any code.

Related: I notice that the webview contents zoom uniformly when the editor zoom changes. Which is great! I change zoom alot, even in a session. But it reminds me that I don't understand why or how that happens. Whatever the mechanism is, is it exposed? Some kind of inheritance, maybe?

TLDR: how to web?
